### PR TITLE
Do not report exposure type

### PR DESCRIPTION
**Pull request by ismailsunni**
### What does it fix?
* Ticket: #3626 
* Funded by: DFAT
* Description: 
  - Add new class for `Do Not Report` that won't be included in the calculation. You can assign it on keyword wizard.

screen shot 2017-07-27 at 11 53 56

screen shot 2017-07-27 at 10 25 50
screen shot 2017-07-27 at 10 26 02

The number above is not added up because of the rounding.

cc @fredychandra

### CODE REVIEW

File: safe/gis/vector/prepare_vector_layer.py [hunk 1]
```diff
@@ -358,6 +358,16 @@ def _remove_features(layer):
                 # i += 1
                 pass
 
+            # If there is exposure that mapped to do not report class
+            if layer_purpose == layer_purpose_exposure['key']:
+                is_do_not_report = layer.keywords['value_map'].get(
+                    'do_not_report')
+                if is_do_not_report:
+                    if feature[field_name] in layer.keywords['value_map'][
+                        'do_not_report']:
```
Comment (ismailsunni) [inline safe/gis/vector/prepare_vector_layer.py]: __[flake8]__
* [E125] continuation line with same indent as next logical line

File: safe/definitions/exposure_classifications.py [hunk 2]
```diff
@@ -671,6 +671,20 @@
                     'link': None
                 }
             ]
+        },
+        {
+            'key': 'do_not_report',
```
Comment (Gustry) [inline safe/definitions/exposure_classifications.py]: can you make this key from a constant : `'key': DO_NOT_REPORT,` ?

File: safe/gis/vector/prepare_vector_layer.py [hunk 3]
```diff
@@ -358,6 +358,16 @@ def _remove_features(layer):
                 # i += 1
                 pass
 
+            # If there is exposure that mapped to do not report class
+            if layer_purpose == layer_purpose_exposure['key']:
+                is_do_not_report = layer.keywords.get('value_map', {}).get(
+                    'do_not_report')
```
Comment (Gustry) [inline safe/gis/vector/prepare_vector_layer.py]: And use this constant here ? `DO_NOT_REPORT` ?
Like this, all our strings are in definitions

**Comment by Gustry** [comment]
Hum, on thing, there is a rule : the `Other` group is always the last one in the classification. Features without classification will go to the `Other` group. But now, if you do that, it will go to`Do Not Report`. I think you need to move it.

**Comment by ismailsunni** [comment]
Yeah, I was expecting it would be difficult. But, it turns out quite easy to add. Thanks

I will move the `Other` to last, and make constant for Do No Report.

**Comment by samnawi** [comment]
Hi @ismailsunni I'm worried if we use the ways like this, it will disappear one of the essences of the landcover analysis.
image
Besides the value of the crop, landcover analysis also generates the affected area for every landcover type. And DM stakeholder still need this number to do some decision.

I think we must separate the ways to set the landcover grouping and to set the field that calculated for crop.

**Comment by Gustry** [comment]
@samnawi I'm not sure I got everything from your message ;-)
The meaning of this `Do Not Report` is to remove some values in your exposure dataset on-the-fly that you don't want to use. We are really removing these features from the exposure in our internal exposure dataset. Is-it not OK?

File: safe/definitions/exposure_classifications.py [hunk 4]
```diff
@@ -29,6 +30,21 @@
 # to generate the value_mapping. Please do not remove it.
 # See https://github.com/kartoza/osm-reporter/wiki how to use this key.
 
+do_not_report_class = {
+    'key': DO_NOT_REPORT,
+    'name': tr('Do Not Report'),
+    'description': tr(
+        'Land use types that will be excluded in the analysis.'),
```
Comment (Gustry) [inline safe/definitions/exposure_classifications.py]: Ah, now you make its own variable. Why not making it generic? (not using `land use`) and to add it to roads/buildings?

Comment (Gustry) [inline safe/definitions/exposure_classifications.py]: Ok, let's keep it simple for now, not to add it other exposures

**Comment by ismailsunni** [comment]
After discussion in gitter (27 July 2017). We decide not to merge this one. There is a problem like @samnawi mention above.

Since the real data of non-crop exposure class won't have productivity/productivity_cost/productivity_value, it won't be shown in the report. Then the problem's solved.

See my comment in https://github.com/inasafe/inasafe/issues/3626#issuecomment-316919070

closing this.

**Pull request closed without merging**